### PR TITLE
Nadabot indexing, registry, provider and stamp

### PR DESCRIPTION
**Pull request by Prometheo**

### CODE REVIEW

**Review by lachlanglen — CHANGES_REQUESTED** [review]
A couple initial requests:

- change `print` statements to `logger.info/warning/error` etc, and convert multiple args to a formatted string (e.g. `logger.info(f"update landing... {data}")`. Import `logger` from `indexer_app.logging` (as opposed to the django/API logger which is in `base.logging`)

- handle exceptions around db transactions

File: nadabot/api.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            else:
+                # Return list
+                serializer = ListSerializer(list)
+                return Response(serializer.data)
+        else:
+            # Return all lists
+            lists = List.objects.all()
+            results = self.paginate_queryset(lists, request, view=self)
+            serializer = ListSerializer(results, many=True)
+            return self.get_paginated_response(serializer.data)
```
Comment (lachlanglen) [inline nadabot/api.py:52]: not relevant to nadabot

File: nadabot/models.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        Account,
+        related_name="registry_id",
+        on_delete=models.CASCADE,
+        primary_key=True,
+        help_text=_("Nadabot registry id."),
+    )
+    owner = models.ForeignKey(
+        Account,
+        on_delete=models.CASCADE,
+        related_name="nadabot",
```
Comment (lachlanglen) [inline nadabot/models.py:24]: perhaps `nadabot_owner`

File: nadabot/models.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        null=False,
+        help_text=_("Registry creation date."),
+    )
+    updated_at = models.DateTimeField(
+        _("updated at"),
+        help_text=_("Registry last update date."),
+    )
+    admins = models.ManyToManyField(
+        Account,
+        related_name="admin_registry",
```
Comment (lachlanglen) [inline nadabot/models.py:44]: i would just make this consistent with the `owner` related_name, e.g. `nadabot_admin_registries`

File: nadabot/models.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        blank=True,
+        null=True,
+        help_text=_("Optional external URL.")
+    )
+    submitted_by = models.CharField(
+        _("submitted by"),
+        max_length=100,
+        null=False,
+        help_text=_("User who submitted this provider.")
+    )
```
Comment (lachlanglen) [inline nadabot/models.py:130]: this should be a ForeignKey to `Account`

File: nadabot/models.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+        _("submitted at (milliseconds)"),
+        null=False,
+        help_text=_("Timestamp of when this provider was submitted.")
+    )
+    stamp_count = models.BigIntegerField(
+        _("stamp count"),
+        default=0,
+        help_text=_("Total number of times this provider has been used successfully."),
+        db_index=True
+    )
```
Comment (lachlanglen) [inline nadabot/models.py:141]: We shouldn't need this as a model field as we can access via `Provider.stamps.count()` and include in serializer if we want

File: nadabot/models.py [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+        Provider,
+        on_delete=models.CASCADE,
+        verbose_name=_("provider"),
+        help_text=_("The provider the user verified with.")
+    )
+    verification_date = models.DateField(
+        _("verification date"),
+        auto_now_add=True,
+        help_text=_("The date of verification.")
+    )
```
Comment (lachlanglen) [inline nadabot/models.py:187]: @Prometheo can you update these models to implement nadabot v2? e.g. include `custom_args` field for `Provider`, add `Group` model

Comment (Prometheo) [reply nadabot/models.py:187]: is nadabot v2 already deployed?

**Comment by lachlanglen** [comment]
Only on staging at this point.

File: nadabot/models.py (hunk 6 above)
Comment (lachlanglen) [reply nadabot/models.py:187]: There's some more info in this ticket: https://github.com/PotLock/nadabot-app/issues/109

Lmk if that clarifies or if you need more info

**Pull request closed without merging**